fc/cli/reports.py

```diff
@@ -0,0 +1,503 @@
+        body=add_sheet).execute()
+
+    cli_library.echo('new sheets for each developer added')
+
+    data_response = service.spreadsheets().values().batchUpdate(
+        spreadsheetId=spreadsheet_id,
+        body=data_body).execute()
+
+    cli_library.echo('data added to sheets')
+
+    sheet_update_requests = []
+
+    sheet_update = {
+        'requests': sheet_update_requests
+    }
+
+    for reply in add_response['replies']:
+        sheet_update_requests.append(build_auto_resize_details(reply['addSheet']['properties']['sheetId'], 0, 7))
+
+    response = service.spreadsheets().batchUpdate(
+        spreadsheetId=spreadsheet_id,
+        body=sheet_update).execute()
+
+    cli_library.echo('columns of all sheets auto-resized')
+
+
+@reports.command()
+@click.option('--username', )
+def vfrsanity(username: str):
+    cli_library.echo('Generating vfr sanity check google sheet')
+
+    auth = ComboAuth(username)
+
+    # setup and build credentials for google api calls
+    g_scopes = None
+    g_service_acct_file = None
+    g_sheet_create_url = None
+
+    (g_scopes, g_service_acct_file, g_sheet_create_url) = build_google_creds(auth)
+
+    if g_service_acct_file is None or g_sheet_create_url is None:
+        cli_library.echo('Google service account credential file path and sheet create url must be defined '
+                         'in order to generate a report file.')
+        return
+
+    cli_library.echo('retrieving all stories')
+
+    # get stories ordered by duration
+    jsonr = tasks.search_for_stories_ord_duration(auth)
+
+    cli_library.echo('story retrieval complete')
+
+    credentials = service_account.Credentials.from_service_account_file(
+        g_service_acct_file, scopes=g_scopes)
+
+    service = build('sheets', 'v4', credentials=credentials)
+
+    report_title = 'VFR Sanity Check {}'.format(date.today().strftime('%m-%d-%y'))
+
+    response = requests.get(g_sheet_create_url + report_title)
+
+    response.raise_for_status()
+
+    response_json = response.json()
+
+    spreadsheet_id = None
+    folder_id = None
+
+    if response_json['success']:
+        spreadsheet_id = response_json['newSheetId']
+        folder_id = response_json['destFolderId']
+        cli_library.echo('successful spreadsheet creation, https://docs.google.com/spreadsheets/d/{}/edit#gid=0, '
+                         .format(spreadsheet_id) +
+                         'folder: {}'.format(folder_id))
+
+    # update the individual sheet name in the spreadsheet we just created
+    sheet_update = {
+        'requests': {
+            'updateSheetProperties': {
+                'properties': {
+                    'title': 'Duration'
+                },
+                'fields': 'title'
+            }
+        }
+    }
+
+    response = service.spreadsheets().batchUpdate(
+        spreadsheetId=spreadsheet_id,
+        body=sheet_update).execute()
+
+    cli_library.echo('default sheet title updated')
```

Comment:
It looks like the below code is roughly split into two sections.  One for the duration and another for the cost of delay.  Those sections could be put into functions.